src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -2,911 +2,591 @@
-            int charCount = jpDecoder.GetChars(shiftJapaneseBuffer, byteIndex, 1, buffer, charIndex);
-            if (charCount > 0)
+            // Adding the code page provider should keep the originals, too.
+            foreach (var mapping in CrossplatformDefaultEncodings().Union(CodePageInfo().Select(i => Map((int)i[0], (string)i[1]))))
             {
-                charIndex += charCount;
+                Encoding encoding = Encoding.GetEncoding(mapping.Key);
+                Assert.NotNull(encoding);
+                Assert.Equal(mapping.Value, encoding.WebName);
             }
-            byteIndex++;
+            // Currently the class EncodingInfo isn't present in corefx, but this method should check for the exhaustive list
+            // When it is, comment out this line and remove the previous foreach/assert.
+            // Assert.Equal(CrossplatformDefaultEncodings().Union(CodePageInfo().Select(i => Map((int)i[0], (string)i[1])).OrderBy(i => i.Key)),
+            //               Encoding.GetEncodings().OrderBy(i => i.CodePage).Select(i => Map(i.CodePage, i.WebName)));
+
+#if !PLATFORM_UNIX
+            Encoding defaultCodePage = Encoding.GetEncoding(0);
+            Assert.True(defaultCodePage.CodePage != 0);
+            Assert.True(defaultCodePage.WebName != Encoding.UTF8.WebName);
+#else
+        // Probably Unix sticks with UTF by default?
+        Assert.Equal(Encoding.UTF8, Encoding.GetEncoding(0));
+#endif
         }
 
-        int[] expectedResult = new int[] { 0x41, 0x42, 0x43, 0x3002, 0x6F86, 0x44, 0x45 };
-        Assert.True(charIndex == expectedResult.Length);
-        for (int i = 0; i < charIndex; i++)
-            Assert.True(expectedResult[i] == (int)buffer[i]);
-    }
-
-    [Fact]
-    public static void TestDecodingIso2022KR()
-    {
-        EnsureInitialization();
-
-        byte[] koreanEncodedBuffer = new byte[]
-            {
-                0x0E, 0x21, 0x7E,
-                0x1B, 0x24, 0x29, 0x43, 0x21, 0x7E,
-                0x0F, 0x21, 0x7E,
-                0x1B, 0x24, 0x29, 0x43, 0x21, 0x7E
-            };
-
-        string codepageName = "iso-2022-kr";
-        Decoder krDecoder = Encoding.GetEncoding(codepageName).GetDecoder();
-        const int BUFFER_SIZE = 100;
-        char[] buffer = new char[BUFFER_SIZE];
-
-        int byteIndex = 0;
-        int charIndex = 0;
-
-        while (byteIndex < koreanEncodedBuffer.Length)
+        [Theory]
+        [MemberData("SpecificCodepageEncodings")]
+        [ActiveIssue(846, PlatformID.AnyUnix)]
+        public static void TestRoundtrippingSpecificCodepageEncoding(string encodingName, byte[] bytes, string expected)
         {
-            int charCount = krDecoder.GetChars(koreanEncodedBuffer, byteIndex, 1, buffer, charIndex);
-            if (charCount > 0)
-            {
-                charIndex += charCount;
-            }
-            byteIndex++;
+            Encoding encoding = CodePagesEncodingProvider.Instance.GetEncoding(encodingName);
```

Comment:
I agree, you can check in TestDefaultCodePage if al encoding exist to know if the other encoding is already registered. that will make the test more reliable and also give us the freedom in the other test cases to register and use the full encodings